test_runner/src/main/kotlin/ftl/shard/TestMethodDuration.kt

```diff
@@ -0,0 +1,45 @@
+package ftl.shard
+
+import ftl.args.AndroidArgs
+import ftl.args.IArgs
+import ftl.reports.xml.model.JUnitTestCase
+import ftl.reports.xml.model.JUnitTestResult
+import ftl.util.FlankTestMethod
+
+object TestMethodDuration {
+
+    private fun JUnitTestCase.androidKey(): String {
+        return "class $classname#$name"
+    }
+
+    private fun JUnitTestCase.iosKey(): String {
+        // FTL iOS XML appends `()` to each test name. ex: `testBasicSelection()`
+        // xctestrun file requires classname/name with no `()`
+        val testName = name?.substringBefore('(')
+        return "$classname/$testName"
+    }
+
+    fun createTestMethodDurationMap(junitResult: JUnitTestResult, args: IArgs): Map<String, Double> {
```

Comment:
Its good idea to sort functions from most to less important. `createTestMethodDurationMap` is more important then `androidKey` and `iosKey` cause is public.